src/sections/layer_and_mask_information_section/mod.rs

```diff
@@ -113,13 +113,12 @@ impl LayerAndMaskInformationSection {
 
             layer_records.push(layer_record);
         }
-        let mut groups = KeyIDContainer::with_capacity(groups_count);
+        let mut groups = NamedItems::with_capacity(groups_count);
 
-        let mut range_stack = vec![];
-        let mut nested_level = 0;
+        let mut group_start_stack = vec![];
         // Read each layer's channel image data
         for (idx, layer_record) in layer_records.into_iter().enumerate() {
-            let group_id = if nested_level > 0 {
+            let group_id = if group_start_stack.len() > 0 {
                 Some((groups_count - groups.len() - 1) as u32)
```

Comment:
I'm confused by this line. It reads as "Number of groups minus number of groups minus 1". What is this doing?

Reply:
- Layers are present in reverse order. Then, the first layer record contains last layer, so `group id` will be `totally_number_of_groups - already_seen_groups`.
- Also, ID starts from 0, because it's a index, then -1.

Reply:
Oh I think I'm following.

So this is leading to `GroupDivider::Close` opening the folder and `GroupDivider::Open` closing the folder.

If that's the same - could we just `for (idx, layer_record) in layer_records.into_iter().enumerate().reverse()` instead of having this inversed logic?